### ISSUE TITLE
Problem with autocorrect Style/Tab and Style/IndentationConsistency

### ISSUE THREAD

**Issue by obromios** [bbatsov/rubocop#1928]
When I run rubocop --autocorrect on my project with Style/Tab and Style/IndentationConsistency enabled I get many errors, due to rubocop splitting up words.  A boiled down example file is:
```
require 'spec_helper'
describe ArticlesController do
  render_views
    describe "GET 'index'" do
		    it "returns http success" do
    		end
		describe "admin user" do
			 before(:each) do
 			end
		end
    end
end
```
The output is as follows:
```
require 'spec_helper'
describe ArticlesController do
  render_views
  describe "GET 'index'" do
          it "returns http success" do
          end
          describe "admin user" d    o
           before(:each) d    o
           en    d
      end
  end
end
```
Note the split `do` after the `before(:each)`.  It not only splits up keywords, but symbols, literals, i.e. it seems to do it pretty much everywhere.

**Comment by obromios** [comment]
Hi
I have updated to 0.32, and bundle does indeed show
```Using rubocop 0.32.0```

If I copy the input code from above and paste into a new file, ```temp_spec.rb``` and run

``be rubocop --auto-correct spec/temp_spec.rb```
then I get exactly the same output as reported above, i.e. it breaks the code. Am I doing something wrong here?

**Comment by jonas054** [comment]
@obromios I get the same result. Sorry about that. I didn't check that my fix resolves the original problem. I'll take another look.

**Comment by obromios** [comment]
That's alright, I am sorry I have such messy code.

**Comment by lumeet** [comment]
I stumbled upon a similar issue with `IndentationConsistency`. Here's a shortened version of the failing case:

```ruby
def method
  a = 1
    if something
      if a
        b

      if test.nil?
      end
      end
    end
end
```

**Comment by jonas054** [comment]
Yes, the problem is not directly related to tab characters. Rather it's the fact that cops that deal with alignment typically change more than one offending line in `autocorrect`. They change the alignment of a whole range of lines. If there are other offenses within the realigned line range, then bad things happen. I'm working on a solution.

**Comment by obromios** [comment]
The latest patch has improved things, but I regret I am still having trouble. Here is an example of what is going wrong:

```
require 'spec_helper'
require 'pp'
describe Admin::AccountsController do
  render_views

    describe 'access control' do
      it 'must be signed in' do
        get :index
        response.should redirect_to signin_path
      end
      context 'signed_in' do
        let!(:user) { setup_and_sign_in_user }
        let!(:account) { user.account }
      it 'rejects ordinary user' do
        get :index
        response.should redirect_to root_path
      end
      it 'admin user' do
        user.toggle(:admin)
        get :index
        h1_is_correct 'Admin for Accounts'
        response.should render_template('admin/accounts/index')
      end
      end
    describe 'methods' do
      let!(:user) { setup_and_sign_in_user }
      let!(:account) { user.account }
      before(:each) { user.toggle(:admin) }
      describe :edit do
        it 'should open right page' do
          get :edit, id: user
          h1_is_correct 'Editing Account'
        end
        it 'Sets right value to set_free_days' do
          get :edit, id: account
          assigns(:free_days_left).should eq 7
        end
      end
      describe :update do
        it 'should allow changes' do
          get :update, id: account, account: { paid_up: true, set_free_days: 0 }
          account.reload
          account.paid_up.should be_true
          account.free_period.should eq 7
        end
        it 'should allow free days setting' do
          get :update, id: account, account: { set_free_days: 11 }
          account.reload
          account.free_period.should eq 11
        end
      end
    end
    end
end
```
is ending up as
```
require 'spec_helper'
require 'pp'
describe Admin::AccountsController do
  render_views

  describe 'access control' do
    it 'must be signed in' do
      get :index
      response.should redirect_to signin_path
    end
    context 'signed_in' do
      let!(:user) { setup_and_sign_in_user }
      let!(:account) { user.account }
      it 'rejects ordinary user' d  o
      get :inde  x
      response.should redirect_to root_pat  h
    end
      it 'admin user' d  o
      user.toggle(:admin  )
      get :inde  x
      h1_is_correct 'Admin for Accounts  '
      response.should render_template('admin/accounts/index'  )
    end
    end
    describe 'methods' d  o
    let!(:user) { setup_and_sign_in_user   }
    let!(:account) { user.account   }
    before(:each) { user.toggle(:admin)   }
    describe :edit d  o
      it 'should open right page' d  o
        get :edit, id: use  r
        h1_is_correct 'Editing Account  '
      en  d
      it 'Sets right value to set_free_days' d  o
        get :edit, id: accoun  t
        assigns(:free_days_left).should eq   7
      en  d
    en  d
    describe :update d  o
      it 'should allow changes' d  o
        get :update, id: account, account: { paid_up: true, set_free_days: 0   }
        account.reloa  d
        account.paid_up.should be_tru  e
        account.free_period.should eq   7
      en  d
      it 'should allow free days setting' d  o
        get :update, id: account, account: { set_free_days: 11   }
        account.reloa  d
        account.free_period.should eq 1  1
      en  d
    en  d
  end
  end
end
```
and 
```
module PlansHelper
  def for_student(student)
    " for #{student.name}" if student.present?
  end

    def handicap_country_list
      [['No Handicap', 0], ['Australian', 1], ['United States', 2], ['New Zealand', 3], ['South African', 4], ['United Kingdom', 5], ['Canadian', 6], ['Other', 7]]
    end

    def durationList
      dList = []
    dList[0] = ['0 hours', 0]
    dList[1] = ['1 hour', 60]
    for i in 2..10
      dList[i] = [i.to_s + ' hours', i * 60]
    end
    nExtra = 9
    nStart = dList.size - 1
    for i in 1..nExtra do
      dList[i + nStart] = [(i * 5 + 10).to_s + ' hours', i * 10 * 60]
    end
    dList[dList.size] = [(nExtra * 10).to_s + '+ hours', nExtra + 1]
    dList
    end

    def lesson_list
      (0..20).to_a
    end
end
```
is ending up as
```
module PlansHelper
  def for_student(student)
    " for #{student.name}" if student.present?
  end

  def handicap_country_list
    [['No Handicap', 0], ['Australian', 1], ['United States', 2], ['New Zealand', 3], ['South African', 4], ['United Kingdom', 5], ['Canadian', 6], ['Other', 7]]
  end

  def durationList
    dList = []
    dList[0] = ['0 hours', 0]
    dList[1] = ['1 hour', 60]
    for i in 2..1  0
    dList[i] = [i.to_s + ' hours', i * 60  ]
  end
    nExtra = 9
    nStart = dList.size - 1
    for i in 1..nExtra d  o
    dList[i + nStart] = [(i * 5 + 10).to_s + ' hours', i * 10 * 60  ]
  end
    dList[dList.size] = [(nExtra * 10).to_s + '+ hours', nExtra + 1]
    dList
  end

  def lesson_list
    (0..20).to_a
  end
end
```

**Comment by jonas054** [comment]
@obromios Are you sure? I get exactly the results you describe when I run on `v0.32.0` but on `v0.32.1` and on `master` I see no problems.

**Comment by obromios** [comment]
I will give that a go.

Chris

**Comment by obromios** [comment]
... and it worked. It is strange though, I had checked [rubygems](https://rubygems.org/gems/rubocop/versions/0.32.1) and it showed 0.32.1 as being released.

Anyway, my code now looks beautiful and the test for that particular file all passed. I will now try rubocop --autocorrect on the whole app and get back to you if there is a problem.

Thank you.
